android/src/main/java/com/pagopa/ioreactnativeintegrity/IoReactNativeIntegrityModule.kt

```diff
@@ -1,25 +1,315 @@
+  }
+
+  /**
+   * Preparation step for a [Play Integrity standard API request](https://developer.android.com/google/play/integrity/standard).
+   * It prepares the integrity token provider before obtaining the integrity verdict.
+   * It should be called well before the moment an integrity verdict is needed, for example
+   * when starting the application. It can also be called time to time to refresh it.
+   * The React Native promise is resolved with an empty payload on success, otherwise
+   * it gets rejected when:
+   * - The preparation fails;
+   * - The provided [cloudProjectNumber] format is incorrect.
+   * @param cloudProjectNumber a Google Cloud project number which is supposed to be composed only by numbers (Long).
+   * @param promise the React Native promise to be resolved or rejected.
+   */
+  @ReactMethod
+  fun prepareIntegrityToken(cloudProjectNumber: String, promise: Promise) {
+    try {
+      val cpn = cloudProjectNumber.toLong()
+      val standardIntegrityManager: StandardIntegrityManager =
+        IntegrityManagerFactory.createStandard(reactApplicationContext)
+      standardIntegrityManager.prepareIntegrityToken(
+        StandardIntegrityManager.PrepareIntegrityTokenRequest.builder()
+          .setCloudProjectNumber(cpn)
+          .build()
+      )
+        .addOnSuccessListener { res -> integrityTokenProvider = res; promise.resolve(null) }
+        .addOnFailureListener { ex -> ModuleException.PREPARE_FAILED.reject(promise, Pair(ERROR_USER_INFO_KEY, getExceptionMessageOrEmpty(ex))) }
+    }catch (_: NumberFormatException){
+      ModuleException.WRONG_GOOGLE_CLOUD_PROJECT_NUMBER_FORMAT.reject(promise)
+    }catch (e: Exception){
+      ModuleException.PREPARE_FAILED.reject(promise, Pair(ERROR_USER_INFO_KEY, getExceptionMessageOrEmpty(e)))
+    }
+  }
+
+  /**
+   * Integrity token request step for a [Play Integrity standard API request](https://developer.android.com/google/play/integrity/standard).
+   * It requests an integrity token which is then attached to the request to be protected.
+   * It should be called AFTER [prepareIntegrityToken] has been called and resolved successfully.
+   * The React Native promise is resolved with with the token as payload or rejected when:
+   * - The integrity token request fails;
+   * - The [prepareIntegrityToken] function hasn't been called previously.
+   * @param requestHash a digest of all relevant request parameters (e.g. SHA256) from the user action or server request that is happening.
+   * The max size of this field is 500 bytes. Do not put sensitive information as plain text in this field.
+   * @param promise the React Native promise to be resolved or rejected.
+   */
   @ReactMethod
-  fun multiply(a: Double, b: Double, promise: Promise) {
-    promise.resolve(a * b)
+  fun requestIntegrityToken(requestHash: String?, promise: Promise) {
+    try {
+      val integrityTokenResponse: Task<StandardIntegrityToken> = integrityTokenProvider.request(
```

Comment:
```suggestion
      val integrityTokenResponse = integrityTokenProvider.request(
```

Reply:
Bad habit, fixed in 135cc11cb40627fdc7e63d63a39015e3c5e737bc, thanks.